test-suite/js/src/test/scala/org/scalajs/testsuite/library/UnionTypeTest.scala

```diff
@@ -134,6 +134,19 @@ object UnionTypeTest extends JasmineTest {
       expect(y4.toJSArray).toEqual(js.Array(3, 5))
     }
 
+    it("UndefOr[Union]") {
+      val a: String = ""
+
+      a: UndefOr[String]
+      a: |[Int, String]
```

Comment:
Also, it would be good, if only to follow the shape of the other tests, to use `expect` to check the result of that conversion, for example:
```scala
expect(a: js.Undefor[String]).toBe(a)
```
To that effect, I suggest using `"hello"` instead of `""` for more comprehensible error messages, should the tests fail.